lsp-mode.el

```diff
@@ -5671,20 +5672,24 @@ while PLACEHOLDER?, is either nil or a string suggested by the
 language server as the initial input of a new-name prompt."
   (unless (lsp-feature? "textDocument/rename")
     (error "The connected server(s) doesn't support renaming"))
-  (if (and lsp-rename-use-prepare (lsp-feature? "textDocument/prepareRename"))
-      (when-let ((response
-                  (lsp-request "textDocument/prepareRename"
-                               (lsp--text-document-position-params))))
-        (let* ((bounds (lsp--range-to-region
-                        (if (lsp-range? response)
-                            response
-                          (lsp:prepare-rename-result-range response))))
-               (placeholder
-                (and (not (lsp-range? response))
-                     (lsp:prepare-rename-result-placeholder response))))
-          (cons bounds placeholder)))
-    (when-let ((bounds (bounds-of-thing-at-point 'symbol)))
-      (cons bounds nil))))
+  (or
+   (and lsp-rename-use-prepare (lsp-feature? "textDocument/prepareRename")
+        (if-let ((response (lsp-request "textDocument/prepareRename"
+                                        (lsp--text-document-position-params))))
+            (unless (lsp-get response :defaultBehavior)
+              (let* ((bounds (lsp--range-to-region
+                              (if (lsp-range? response)
+                                  response
+                                (lsp:prepare-rename-result-range response))))
+                     (placeholder
+                      (and (not (lsp-range? response))
+                           (lsp:prepare-rename-result-placeholder response))))
+                (cons bounds placeholder)))
+          ;; If prepareRename returned null, fail now and don't fall back to the
+          ;; symbol at point.
+          (cl-return nil)))
```

Comment:
after you have deleted the check in lsp-rename this should be `user-error` - unable to rename symbol or something like that. This is a regression from the previous PR.

Reply:
The `user-error` happens in `lsp--read-rename`, to which the result of this function is passed unmodified.